firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -124,6 +134,51 @@ public Thread newThread(Runnable r) {
     this.utils = utils;
     this.iidStore = iidStore;
     this.fidGenerator = fidGenerator;
+    preventParallelUsageOfIncompatibleVersionOfInstanceIdSDk();
+  }
+
+  private void preventParallelUsageOfIncompatibleVersionOfInstanceIdSDk() {
+    if (!isIIDVersionCompatible()) {
+      throw new IllegalStateException(
+          "FirebaseInstallations will not work correctly with current version of Firebase "
+              + "Instance ID SDK. Please update your Firebase Instance ID version.");
+    }
+  }
+
+  /**
+   * Returns TRUE iff IID SDK is not used along with FIS SDK or IID SDK version is &gt than 20.1.1,
+   * otherwise returns FALSE.
+   */
+  private boolean isIIDVersionCompatible() {
+    try {
+      final Class<FirebaseInstanceId> firebaseInstanceIdClass = FirebaseInstanceId.class;
```

Comment:
How about using the fully qualified class path here?
com.google.firebase.iid.FirebaseInstanceId.class